hw3/01-routing.js

```diff
@@ -14,38 +14,77 @@ const port = process.env.PORT || 5000;
 // For other routes, such as http://localhost:5000/other, this exercise should return a status code 404 with '404 - page not found' in html format
 
 const server = http.createServer((req, res) => {
-  const routes = [
-    'welcome',
-    'redirect',
-    'redirected',
-    'cache',
-    'cookie',
-    'check-cookies',
-    'other',
-  ];
-
-  let getRoutes = () => {
-    let result = '';
-
-    routes.forEach(
-      (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`)
-    );
-
-    return result;
-  };
-
-  if (req.url === '/') {
-    let routeResults = getRoutes();
-
-    res.writeHead(200, { 'Content-Type': 'text/html' });
-    res.write(`<h1>Exercise 01</h1>`);
-    res.write(`<ul> ${routeResults} </ul>`);
-    res.end();
-  }
-
-  // Add your code here
-});
+    const routes = [
+        'welcome',
+        'redirect',
+        'redirected',
+        'cache',
+        'cookie',
+        'check-cookies',
+        'other',
+    ];
+
+    const getRoutes = () => {
+        let result = '';
+
+        routes.forEach((elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`));
 
+        return result;
+    };
+
+    if (req.url === '/') {
+        let routeResults = getRoutes();
+        res.writeHead(200, { 'Content-Type': 'text/html' });
+        res.write(`<h1>Exercise 01</h1>`);
+        res.write(`<ul> ${routeResults} </ul>`);
+        res.end();
+    } else if (req.url === '/welcome') {
+        res.writeHead(200, { 'Content-Type': 'text/html' });
+        res.write("<H1>Welcome to Prachi's page!</H1>");
+        res.end();
+    } else if (req.url === '/redirect') {
+        res.writeHead(302, { Location: '/redirected' });
+        res.end();
+    } else if (req.url === '/redirected') {
+        res.writeHead(200, { 'Content-Type': 'text/html' });
+        res.write('<H1>You just got redirected!</H1>');
+        res.end();
+    } else if (req.url === '/cache') {
+        res.writeHead(200, {
+            'Content-Type': 'text/html',
+            'Cache-Control': 'max-age = ' + 24 * 60 * 60,
+        });
+        res.write('<H1>This resource was cached!</H1>');
+        res.end();
+    } else if (req.url === '/cookie') {
+        res.writeHead(200, {
+            'Content-Type': 'text/plain',
+            'Set-Cookie': 'hello=world',
+        });
+        res.write('<H1>Cookies...Yummm!<H1>');
+        res.end();
+    } else if (req.url === '/check-cookies') {
+        let message = 'no';
+        let cookie = decodeURIComponent(req.headers.cookie);
+        let cookie_list = cookie.split(';');
+
+        cookie_list.every((element) => {
```

Comment:
Array.prototype.every asks a universal quantified question about the array. It returns a boolean depending on if every item in the array satisifies the condition expressed in the callback function. You're not capturing this value here, only using the iteration the method provides and performing a side effect. If you are just using this method for the iteration, i'd use forEach. However there is a method that does exactly what you want without side effects:
Array.prototype.some

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
This isn't a necessary change, just background info. Sorry if I explained too much, I'm just interested in this stuff

Reply:
@ethanSE No issues, it's great to have information of any type. Will look into it and try to make changes accordingly.

Reply:
@ethanSE great explanation!